README.md

```diff
@@ -1,2 +1,83 @@
 # mplex
-This repo contains the spec of mplex, the friendly Stream Multiplexer (that works in 3 languages!)
+
+> This repo contains the spec of mplex, the friendly Stream Multiplexer (that works in 3 languages!)
+
+mplex is a Stream Multiplexer protocol used by js-ipfs and go-ipfs in their implementations. The origins of this protocol are based in [multiplex](https://github.com/maxogden/multiplex), the JavaScript only Stream Multiplexer. After many battle field tests, we felt the need to improve and fix some of its bugs and mechanics, resulting on this new version used by libp2p.
+
+This document will attempt to define a specification for the wire protocol and algorithm used in both implementations. 
+
+Multiplex is a very simple protocol that does not provide many features offered by other stream multiplexers. Notably, multiplex does not provide backpressure at the protocol level, or support half closed streams.
+
+Implementations in:
+
+- [JavaScript](https://github.com/libp2p/js-libp2p-mplex)
+- [Go](https://github.com/whyrusleeping/go-multiplex)
+- [Rust](https://github.com/libp2p/rust-libp2p/tree/master/multiplex-rs)
+
+## Message format
+
+Every communication in multiplex consists of a header, and a length prefixed data segment.
+
+The header is an unsigned base128 varint, as defined in the [protocol buffers spec](https://developers.google.com/protocol-buffers/docs/encoding#varints). The lower three bits are the message flags, and the rest of the bits (shifted down by three bits) are the stream ID this message pertains to:
+
+```
+header = readUvarint()
+flag = head & 0x07
+id = flag >> 3
+```
+
+### Flag Values
+
+| NewStream        | 0 |
+| MessageReceiver  | 1 |
+| MessageInitiator | 2 |
+| CloseReceiver    | 3 |
+| CloseInitiator   | 4 |
+| ResetReceiver    | 5 |
+| ResetInitiator   | 6 |
```

Comment:
Not displaying correctly.  Try adding a header to the table, e.g. `| Flag | Value| \n | ---- | ------ |`

Reply:
I fixed it in master.